BitwardenShared/Core/Platform/Services/AuthenticatorSyncService.swift

```diff
@@ -0,0 +1,247 @@
+        self.notificationCenterService = notificationCenterService
+        self.stateService = stateService
+        self.vaultTimeoutService = vaultTimeoutService
+        super.init()
+    }
+
+    // MARK: Public Methods
+
+    public func start() {
+        Task {
+            if await configService.getFeatureFlag(FeatureFlag.enableAuthenticatorSync,
+                                                  defaultValue: false) {
+                subscribeToAppState()
+            }
+        }
+    }
+
+    // MARK: Private Methods
+
+    /// Check to see if the shared Authenticator key exists. If it doesn't already exist, create it.
+    ///
+    private func createAuthenticatorKeyIfNeeded() async throws {
+        let storedKey = try? await sharedKeychainRepository.getAuthenticatorKey()
+        if storedKey == nil {
+            let key = SymmetricKey(size: .bits256)
+            let data = key.withUnsafeBytes { Data(Array($0)) }
+            try await sharedKeychainRepository.setAuthenticatorKey(data)
+        }
+    }
+
+    /// Take a list of encrypted ciphers, decrypt them, filter for only active ciphers with a totp code, then
+    /// convert the list to AuthenticatorSyncItemDataModel to be stored and sync'd to the Authenticator app.
+    ///
+    /// - Parameters:
+    ///   - ciphers: The encrypted `Cipher` objects.
+    ///   - userId: The userId of the account to which these Ciphers belong.
+    ///
+    /// - Returns: The decrypted, filtered, and sorted `CipherDTO` objects.
+    ///
+    private func decryptTOTPs(_ ciphers: [Cipher],
+                              userId: String) async throws -> [AuthenticatorBridgeItemDataView] {
+        let decryptedCiphers = try await ciphers.asyncMap { cipher in
+            try await self.clientService.vault().ciphers().decrypt(cipher: cipher)
+        }
+        let account = try await stateService.getActiveAccount()
+        let username = account.profile.name ?? account.profile.email
+
+        let ciphersToUse = decryptedCiphers.filter { cipher in
+            cipher.deletedDate == nil
+                && cipher.type == .login
+                && cipher.login?.totp != nil
+        }
+
+        return ciphersToUse.map { cipher in
+            AuthenticatorBridgeItemDataView(
+                favorite: false,
+                id: cipher.id ?? UUID().uuidString,
+                name: cipher.name,
+                totpKey: cipher.login?.totp,
+                username: username
+            )
+        }
+    }
+
+    /// This function handles the initial syncing with the Authenticator app as well as listening for updates
+    /// when the user adds new items. This is called when the sync is turned on.
+    ///
+    /// - Parameter userId: The userId of the user who has turned on sync.
+    ///
+    private func handleSyncOnForUserId(_ userId: String) async {
+        Logger.application.log("#### sync is on for userId: \(userId)")
+
+        if !vaultTimeoutService.isLocked(userId: userId) {
```

Comment:
⛏️ Could we use `guard` to reduce nesting?

Reply:
👍 Fixed in latest.